checks/__init__.py

```diff
@@ -375,6 +384,36 @@ def __init__(self, name, init_config, agentConfig, instances=None):
             self.proxies['http'] = "http://{uri}".format(uri=uri)
             self.proxies['https'] = "https://{uri}".format(uri=uri)
 
+    def _get_check_manifest(self, manifest_path=None):
+        """
+        Attempts to collect the manifest for SDK checks.
+
+        Do not use outside of the constructor if called without args, it assumes the following call chain:
+            CheckClass.__init__() -> AgentCheck.__init__() -> AgentCheck._get_check_manifest()
+        """
+        from checks.network_checks import NetworkCheck
+        if not manifest_path:
+            frames = inspect.stack()
+            frame_idx = NETWORK_CHECK_FRAME if isinstance(self, NetworkCheck) else AGENT_CHECK_FRAME
+            caller_frame = frames[frame_idx]  # frame_idx _should_ be the check __init__ frame
+            module_path = os.path.dirname(caller_frame[1])  # idx 1 contains the filename.
+            manifest_path = os.path.join(module_path, 'manifest.json')
+
+        return load_manifest(manifest_path)
+
+    def set_check_version(self, manifest=None):
+        version = AGENT_VERSION
+
+        if manifest:
+            try:
+                version = "{core}:{sdk}".format(core=AGENT_VERSION,
+                                            sdk=manifest['version'])
+            except Exception:
+                self.log.info('Invalid manifest - unable to extract check version.')
+                version = 'Unknown'
```

Comment:
Well, because `AGENT_VERSION` represents the core agent shipped checks. That wouldn't be the case here, it would be an SDK check with an unknown version.